core/src/main/java/io/kestra/core/runners/ExecutorService.java

```diff
@@ -717,19 +736,53 @@ private Executor handleWorkerTask(Executor executor) throws InternalException {
             .map(throwFunction(taskRun -> {
                 Task task = executor.getFlow().findTaskByTaskId(taskRun.getTaskId());
                 RunContext runContext = runContextFactory.of(executor.getFlow(), task, executor.getExecution(), taskRun);
-                return WorkerTask.builder()
+                WorkerTask workerTask = WorkerTask.builder()
                     .runContext(runContext)
                     .taskRun(taskRun)
                     .task(task)
                     .build();
-            }))
+                String workerGroup = workerGroupService.resolveGroupFromJob(workerTask);
+                // Check if the worker group exist
+                if (workerGroupService.isWorkerGroupExistForKey(workerGroup)) {
+                    // Check whether at-least one worker is available
+                    if (workerGroupService.isWorkerGroupAvailableForKey(workerGroup)) {
+                        return workerTask;
+                    } else {
+                        runContext.logger()
+                            .error("Cannot run task. No workers are available for worker group '" + workerGroup + "'.");
+                    }
+                } else {
+                    runContext.logger()
+                        .error("Cannot run task. No worker group exist for key '" + workerGroup + "'.");
+                }
+                // fail the task-run because no worker can run the task
+                return workerTask.withTaskRun(workerTask.getTaskRun()
+                    .withState(State.Type.FAILED)
+                    .withAttempts(Collections.singletonList(
+                        TaskRunAttempt.builder().state(new State().withState(State.Type.FAILED)).build()
+                    ))
+                );
+            })
+            )
             .toList();
 
         if (workerTasks.isEmpty()) {
             return executor;
         }
 
-        return executor.withWorkerTasks(workerTasks, "handleWorkerTask");
+        List<WorkerTaskResult> failed = workerTasks
+            .stream()
+            .filter(workerTask -> workerTask.getTaskRun().getState().isFailed())
+            .map(workerTask -> WorkerTaskResult.builder().taskRun(workerTask.getTaskRun()).build())
+            .toList();
+
+        if (failed.isEmpty()) {
+            return executor.withWorkerTasks(workerTasks, "handleWorkerTask");
+        } else {
+            return executor
+                .withWorkerTasks(workerTasks, "handleWorkerTask")
+                .withWorkerTaskResults(failed, "handleWorkerTask");
+        }
```

Comment:
Is it really needed to add fake workerTaskResult? As you directly add a failed taskrun to the worker task this should not be needed.

Or the opposite should not be needed, added a failed worker task run to the worker task as this will be done automatically but the new worker task result.